### DISCUSSION TITLE
Problem with French parsing when using apostrophe

### DISCUSSION THREAD

**Discussion by pastrop** [explosion/spaCy#9411] [Language Support]
Hi I just came across a strange problem: while parsing "c'est la vie" using docker container at https://spacy.io/models/fr I am getting a correct result:
C'est la vie
C' ce PRON nsubj
est être AUX cop
la le DET det
vie vie NOUN ROOT

BUT if I try to do the same in my code downloading the latest spaCy version:
{'location': '/usr/local/lib/python3.7/dist-packages/spacy',
 'pipelines': {'fr_core_news_sm': '3.0.0'},
 'platform': 'Linux-5.4.104+-x86_64-with-Ubuntu-18.04-bionic',
 'python_version': '3.7.12',
 'spacy_version': '3.1.3'}

I am getting the following:
C'est c'est NOUN 
la le DET 
vie vie NOUN

The latter is obviously wrong.  I would appreciate to understand why it is happening.

**Comment by polm** [comment]
Hello, I can't seem to reproduce this. Couple of things to check:

1. I assume it's the same as the code at the link, but please share your code.
2. Try downloading the French model again. Your model is not the latest version, which is 3.1, not 3.0. (I also tried the 3.0 model, so I don't think that's the issue, but it's worth upgrading anyway.)
3. Are you sure the apostrophe is an ASCII apostrophe and not some other character? I tried a `U+2019 ’ RIGHT SINGLE QUOTATION MARK` and that changed the output but not the tokenization.

**Reply by pastrop** [reply]
Hello, After trying everything I could think of, I think I found some sort of a bug with spaCy 3.x French models!  Please read till the end. So here is exactly what I have done.
1. I started with the latest version of the model ('spacy_version': '3.1.3') 
2. get the french model via: spacy.cli.download('fr_core_news_sm') 
3. got the following message: /usr/local/lib/python3.7/dist-packages/spacy/util.py:758: UserWarning: [W095] Model 'fr_core_news_sm' (3.0.0) was trained with spaCy v3.0 and may not be 100% compatible with the current version (3.1.3). If you see errors or degraded performance, download a newer compatible model or retrain your custom model with the current spaCy version. For more details and available updates, run: python -m spacy validate
  warnings.warn(warn_msg)
4. Ignored the warning, parsed my test phrase: "C'est la vie" and got:
C'est c'est NOUN
la le DET
vie vie NOUN
5. So I went to https://spacy.io/models/fr verified that everything is parsing correctly using. the try-out  sandbox available there and check the versions that are used there: I got: 
{'spacy_version': '3.0.7', 'location': '/srv/conda/envs/notebook/lib/python3.7/site-packages/spacy', 'platform': 'Linux-5.4.109+-x86_64-with-debian-buster-sid', 'python_version': '3.7.10', 'pipelines': {'nb_core_news_sm': '3.0.0', 'ro_core_news_sm': '3.0.0', 'nl_core_news_sm': '3.0.0', 'zh_core_web_sm': '3.0.0', 'fr_core_news_sm': '3.0.0', 'en_core_web_md': '3.0.0', 'ru_core_news_sm': '3.0.0', 'en_core_web_sm': '3.0.0', 'el_core_news_sm': '3.0.0', 'pt_core_news_sm': '3.0.0', 'da_core_news_sm': '3.0.0', 'es_core_news_sm': '3.0.0', 'pl_core_news_sm': '3.0.0', 'de_core_news_sm': '3.0.0', 'it_core_news_sm': '3.0.0', 'mk_core_news_sm': '3.0.0', 'lt_core_news_sm': '3.0.0', 'ja_core_news_sm': '3.0.0'}} 
6. I downgraded my spaCy version to 3.0.7, reinstalled again using spacy.cli.download('fr_core_news_sm') and again got:
C'est c'est NOUN
la le DET
vie vie NOUN

I tried it both inside my code and as an isolated test in Google Colab and here is the kicker: I know that Google Colab has the older version of the spaCy installed (2.2.4).  So I wondered what would happen if I use spaCy 2.2.4?  So I simply import to default Google Colab spaCy model (2.2.4), again run spacy.cli.download('fr_core_news_sm') to get the French model and parsed "C'est la vie".  I got:
C' ce PRON PRON__Number=Sing|Person=3
est être AUX AUX__Mood=Ind|Number=Sing|Person=3|Tense=Pres|VerbForm=Fin
la le DET DET__Definite=Def|Gender=Fem|Number=Sing|PronType=Art
vie vie NOUN NOUN__Gender=Fem|Number=Sing
That is correct!  So, there is something weird going in with spaCy 3.x series and the French model.  again, whatever runs in the docker container used by the trial sandbox on the spaCy website works fine!

**Reply by polm** [reply]
Thanks for spending extra time investigating this, but I think you're misunderstanding some things and the information you've given doesn't help with reproducing things.

> I started with the latest version of the model ('spacy_version': '3.1.3')

That's the version of the spaCy library, not the model. For the v3 models the only versions are `3.0.0` or `3.1.0`, like you see in the output from the demo page. 

If your spaCy version is 3.1.3 and you download a model, you should get the 3.1.0 model and you shouldn't get that W095 warning message. I'm not sure how could happen. Are you sure you downloaded the model after upgrading spaCy?

However, when I test locally with that same configuration you say had issues - spaCy v3.1.3 and fr_core_news_sm v3.0.0 and the W095 warning - I still don't get the output you get. I'm not sure what the difference is there.

**Reply by svlandeg** [reply]
@pastrop : to rule out any issues because of incompatible (older) versions, could you try this again from a clean environment?
```
python -m venv .env
source .env/bin/activate
```

**Reply by pastrop** [reply]
As of now,  expression with apostrophe (i.e. "c'est") are parsed correctly using 'spacy_version': '3.1.3' and model version 3.1.0.  I just made it all work in Google Colab. Over the weekend, I couldn't make it work either in Colab or locally using the same combination of versions.  I will let you know if I see anything weird after I update the models in my code.
